### PR TITLE
new implementations for variance and stdev reductions

### PR DESCRIPTION
**Pull request by quickwritereader**
## What changes were proposed in this pull request?

new implementations are used for variance,stdev reductions. minor fix
in stdev. Bench and correctness tests are moved inside #ifdef block

## How was this patch tested?

unit tests

## Quick checklist

The following checklist helps ensure your PR is complete:

- [x] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.org/eclipse-contributors) page for details
- [x] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [x] Created tests for any significant new code additions.
- [x] Relevant tests for your changes are passing.

### CODE REVIEW

**Comment by quickwritereader** [comment]
eclipse/deeplearning4j#8826

File: libnd4j/blas/CMakeLists.txt [hunk 1]
```diff
@@ -300,7 +300,7 @@ elseif(SD_CPU)
                 string(REGEX MATCH "^(.*)\\.cpp\.in$" dummy ${FL_ITEM})
                 set(FL_ITEM_WLE ${CMAKE_MATCH_1})
                 foreach(FL_TYPE_INDEX RANGE 0 9)
-                    #message( "${FL_ITEM_WLE}_${FL_TYPE_INDEX}.cpp")
+                     message( "${FL_ITEM_WLE}_${FL_TYPE_INDEX}.cpp")
```
Comment (raver119) [inline libnd4j/blas/CMakeLists.txt]: Please, do not keep this message printed.

File: libnd4j/blas/CMakeLists.txt [hunk 2]
```diff
@@ -287,7 +287,7 @@ elseif(SD_CPU)
     file(GLOB_RECURSE MEMORY_SOURCES false ../include/memory/*.cpp ../include/memory/*.h)
     file(GLOB_RECURSE GRAPH_SOURCES false ../include/graph/*.cpp ../include/graph/*.h)
     file(GLOB_RECURSE CUSTOMOPS_SOURCES false ../include/ops/declarable/generic/*.cpp)
-    file(GLOB_RECURSE CUSTOMOPS_GENERIC_SOURCES false ../include/ops/declarable/helpers/cpu/*.cpp ../include/ops/declarable/helpers/impl/*.cpp)
+    file(GLOB_RECURSE CUSTOMOPS_GENERIC_SOURCES false ../include/ops/declarable/helpers/cpu/*.cpp ../include/ops/declarable/helpers/impl/*.cpp ../include/ops/declarable/helpers/cpu/*.hpp)
```
Comment (raver119) [inline libnd4j/blas/CMakeLists.txt]: Why would we need hpp here?

File: libnd4j/include/ops/ops.h [hunk 3]
```diff
@@ -4314,6 +4314,9 @@ namespace simdOps {
 	public:
 
         static _CUDA_HD inline Z getValue(const bool biasCorrected, functions::summarystats::SummaryStatsData<X> val) {
+#if 0
+			nd4j_printf("%d %f %f %f\n", biasCorrected, val.n, val.mean, val.M2);
```
Comment (raver119) [inline libnd4j/include/ops/ops.h]: Please don't keep such messages.

**Review by raver119 — CHANGES_REQUESTED** [review]
Few changes required + there's no CUDA implementation at this moment.

**Pull request closed without merging**